app/src/test/java/com/google/android/ground/ui/offlineareas/selector/OfflineAreaSelectorFragmentTest.kt

```diff
@@ -64,4 +86,45 @@ class OfflineAreaSelectorFragmentTest : BaseHiltTest() {
         matches(hasDescendant(withText(fragment.getString(R.string.offline_area_selector_title))))
       )
   }
+
+  @Test
+  fun `stopDownloading cancels active download and updates UI state`() = runWithTestDispatcher {
+    composeTestRule.setContent { DownloadProgressDialog(viewModel.downloadProgress.value!!, {}) }
```

Comment:
Instead of manually inflating it, can we click on the "Download" button?

Reply:
the download button activation is based on map, hence did this.

Reply:
We can manually enable the button using `downloadButtonEnabled`

Reply:
As discussed, let's add a TODO and resolve this in a separate PR as the current test setup skips a lot of user flows.